src/semantic_release/version/version.py

```diff
@@ -167,11 +202,30 @@ def is_prerelease(self) -> bool:
 
     def __str__(self) -> str:
         full = f"{self.major}.{self.minor}.{self.patch}"
-        prerelease = (
-            f"-{self.prerelease_token}.{self.prerelease_revision}"
-            if self.prerelease_revision
-            else ""
-        )
+        if self.version_compat == "semver":
+            pre_str = f"-{self.prerelease_token}.{self.prerelease_revision}"
+        elif self.version_compat == "pep440":
```

Comment:
Since you are tranforming the prerelease_token in a lossy way the --print_tag command line option may not actually represent the actual tag that will be stored in the repo.  The print_tag takes the str(version) and converts it to a tag.  I would not not do the transformation and keep 'alpha', 'beta', ... since they are allowed by the pep440.

We should implement unit test for this case.

Reply:
If this being done for normalization for sorting we should make a normalize method on version just for sorting.

Reply:
> Since you are tranforming the prerelease_token in a lossy way the --print_tag command line option may not actually represent the actual tag that will be stored in the repo. The print_tag takes the str(version) and converts it to a tag. 

This is a good catch but actually unrelated to this PR. This is a bug and will be fixed separately. It is supposed to provide the actual tag that would match the repo. An e2e test will be written for this bug fix.

Reply:
Yes the planned implementation is to evaluate the repo for all release tags and convert them to Version Objects. First we try the SEMVER parser and then secondly we try a pep440 parser to normalize all versions that match a tag format.  This way sorting is correct regardless of what version spec they used.  

The resulting tag or printed version (and/or tag) is where the `--pep440` parameter and a configuration option tweak PSR to output a pep440 formatted version.

Reply:
> This is a good catch but actually unrelated to this PR. This is a bug and will be fixed separately

That fix still converts the version to a string so this transformation when converting to a string will still break producing the next tag.  The only reason to normalize the version is for sorting.